### PR TITLE
EOS-13953:s3server deployment failed

### PR DESCRIPTION
**Pull request by sradhanand-pati**
Signed-off-by: sradhanand.pati <sradhanand.pati@seagate.com>

### CODE REVIEW

File: srv/components/ha/haproxy/config.sls [hunk 1]
```diff
@@ -58,5 +58,10 @@ Setup haproxy config to enable logs:
     - makedirs: True
     - keep_source: False
 
+Stop haproxy service:
```
Comment (ypise) [inline srv/components/ha/haproxy/config.sls:61]: No
Just add
`- components.ha.haproxy.stop ` in init.sls

File: srv/components/ha/haproxy/init.sls [hunk 2]
```diff
@@ -20,5 +20,7 @@ include:
   - components.ha.haproxy.prepare
   - components.ha.haproxy.install
   - components.ha.haproxy.config
+{%- if "physical" not in grains['virtual'] %}
```
Comment (ypise) [inline srv/components/ha/haproxy/init.sls:23]: No start in this file. Add start in deploy-vm

File: srv/components/s3server/start.sls [hunk 3]
```diff
@@ -19,9 +19,11 @@ include:
   - components.misc_pkgs.rsyslog.stop
   - components.misc_pkgs.rsyslog.start
 
+{%- if "physical" not in grains['virtual'] %}
```
Comment (ypise) [inline srv/components/s3server/start.sls:22]: Remove this block

File: srv/components/ha/haproxy/files/haproxy.cfg [hunk 4]
```diff
@@ -98,8 +98,10 @@ defaults
 #---------------------------------------------------------------------
 frontend main
     # s3 server port
+{%- if pillar['cluster']['cluster_ip'] %}
```
Comment (ypise) [inline srv/components/ha/haproxy/files/haproxy.cfg:101]: Also, check if `cluster_ip` is available before we set it.

**Review by ypise — CHANGES_REQUESTED** [review]

File: srv/components/ha/haproxy/config.sls (hunk 1 above)
Comment (sradhanand-pati) [reply srv/components/ha/haproxy/config.sls]: Done

File: srv/components/s3server/start.sls (hunk 3 above)
Comment (sradhanand-pati) [reply srv/components/s3server/start.sls]: Done

File: srv/components/ha/haproxy/files/haproxy.cfg (hunk 4 above)
Comment (sradhanand-pati) [reply srv/components/ha/haproxy/files/haproxy.cfg]: Done

**Review by ypise — APPROVED** [review]

**Pull request merged**